components/config/opts.rs

```diff
@@ -1065,15 +1058,9 @@ pub fn from_cmdline_args(args: &[String]) -> ArgumentParsingResult {
     }
 
     if let Some(layout_threads) = layout_threads {
-        PREFS.set("layout.threads", PrefValue::Number(layout_threads as f64));
-    } else if let Some(layout_threads) = PREFS.get("layout.threads").as_string() {
-        PREFS.set(
-            "layout.threads",
-            PrefValue::Number(layout_threads.parse::<f64>().unwrap()),
-        );
-    } else if *PREFS.get("layout.threads") == PrefValue::Missing {
-        let layout_threads = cmp::max(num_cpus::get() * 3 / 4, 1);
-        PREFS.set("layout.threads", PrefValue::Number(layout_threads as f64));
+        prefs::pref_map()
+            .set("layout.threads", layout_threads as i64)
+            .expect("Could not set preference: layout.threads");
```

Comment:
I have realised that I intended to replace these cases with `set_pref!`, to avoid using the string names except where that is essential. I will make that change now.